Functions.Templates/Templates/HttpTrigger-Python/__init__.py

```diff
@@ -4,21 +4,19 @@
 
 
 def main(req: func.HttpRequest) -> func.HttpResponse:
-    logging.info('Python HTTP trigger function processed a request.')
+    logging.info("Python HTTP trigger function processed a request.")
 
-    name = req.params.get('name')
-    if not name:
-        try:
-            req_body = req.get_json()
-        except ValueError:
-            pass
-        else:
-            name = req_body.get('name')
+    try:
+        # Get value for `name` from parameters or JSON body
+        name = req.params.get("name") or req.get_json().get("name")
+    except ValueError:

```

Comment:
What is throwing `ValueError`? If `req.params` is a dict then that won't do it. Is it because `req.get_json()` might not exist? If the latter is true then you could also do:
```python
name = req.params.get("name") or getattr(req, "get_json", lambda: {})().get("name")
```
Not sure if that's really readable, but otherwise the `try` block seems overly broad.

Reply:
It's because `get_json()` might not exist. I agree about the `try` block. I was hoping to avoid it if possible.

Reply:
You can see the error being raised here: https://github.com/Azure/azure-functions-python-worker/blob/dev/azure/functions_worker/bindings/http.py

Reply:
Lines 49 through 59

Reply:
```python
name = req.params.get("name")  # If this was Python 3.8 you could use assignment expressions 😁 .
if not name:
    try:
        json_data = req.get_json()
    except AttributeError:  # You had ValueError, but you said "might not exist".
        pass
    else:
        name = json_data.get("name")
```
Or:
```python
name = req.params.get("name")
if not name and hasattr(req, "get_json"):  # Once again, assignment expression would be nice. 😁 
    name = req.get_json().get("name")
```

Reply:
I like the latter. I will implement that one.

Reply:
Reopening. While testing this, I found that `hasattr(req, "get_json")` is always `True`, even when the request doesn't have JSON.